### PR TITLE
[KYUUBI #5295] Use proxy user to kill application on YARN

### LINKED ISSUE #5295: [Feature] Support use proxy user to kill application on YARN
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

As title,

When user call deleteBatch or just call ApplicationOperation kill application, may face real user(which kyuubi server used for kerberos) have no premission to kill application on yarn. 

### Motivation

As describe

### Describe the solution

Pass proxy user to kill application

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

As title,

See more detail in close #5295 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

**Comment by pan3793** [comment]
@zhouyifan279

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala [hunk 1]
```diff
@@ -63,7 +65,18 @@ class YarnApplicationOperation extends ApplicationOperation with Logging {
...
           try {
             val applicationId = reports.get(0).getApplicationId
-            yarnClient.killApplication(applicationId)
+            appMgrInfo.proxyUser match {
+              case Some(proxyUser) =>
+                UserGroupInformation
+                  .createProxyUser(proxyUser, UserGroupInformation.getCurrentUser)
+                  .doAs(new PrivilegedExceptionAction[Unit]() {
+                    override def run(): Unit = {
+                      yarnClient.killApplication(applicationId)
```
Comment (zhouyifan279) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala:74]: UGI of `yarnClient` is set when we call `yarnClient.start()`.  `yarnClient.killApplication` will always run as the UGI set at start.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala [hunk 2]
```diff
@@ -63,7 +65,18 @@ class YarnApplicationOperation extends ApplicationOperation with Logging {
         } else {
           try {
             val applicationId = reports.get(0).getApplicationId
-            yarnClient.killApplication(applicationId)
+            appMgrInfo.proxyUser match {
```
Comment (zhouyifan279) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala:68]: We should still kill applications using kyuubi server UGI, by default.

**Comment by pan3793** [comment]
Close and in favor https://github.com/apache/kyuubi/pull/5306

**Pull request closed without merging**